src/test/scala/com/criteo/marathon/CapabilitiesPluginSpec.scala

```diff
@@ -0,0 +1,209 @@
+            CapabilityInfo.Capability.NET_ADMIN
+          ).asJava)
+        }
+      }
+
+      describe("Effective capabilities") {
+        it("should contain one effective capability") {
+          val plugin = new CapabilitiesPlugin
+          val appSpec = mock[ApplicationSpec]
+          val taskInfoBuilder = TaskInfo.newBuilder
+
+          taskInfoBuilder.setCommand(CommandInfo.newBuilder()
+            .setValue("./a_command"))
+          taskInfoBuilder.getLabelsBuilder.addLabels(Label.newBuilder()
+            .setKey("MESOS_EFFECTIVE_CAPABILITIES")
+            .setValue("NET_RAW"))
+
+          plugin.taskInfo(appSpec, taskInfoBuilder)
+
+          assert(taskInfoBuilder.getContainer.getLinuxInfo.getEffectiveCapabilities.getCapabilitiesList == List(
+            CapabilityInfo.Capability.NET_RAW
+          ).asJava)
+        }
+
+        it("should contain two effective capabilities") {
+          val plugin = new CapabilitiesPlugin
+          val appSpec = mock[ApplicationSpec]
+          val taskInfoBuilder = TaskInfo.newBuilder
+
+          taskInfoBuilder.setCommand(CommandInfo.newBuilder()
+            .setValue("./a_command"))
+          taskInfoBuilder.getLabelsBuilder.addLabels(Label.newBuilder()
+            .setKey("MESOS_EFFECTIVE_CAPABILITIES")
+            .setValue("NET_RAW;NET_ADMIN"))
+
+          plugin.taskInfo(appSpec, taskInfoBuilder)
+
+          assert(taskInfoBuilder.getContainer.getLinuxInfo.getEffectiveCapabilities.getCapabilitiesList == List(
+            CapabilityInfo.Capability.NET_RAW,
+            CapabilityInfo.Capability.NET_ADMIN
+          ).asJava)
+        }
+      }
+
+      describe("Capabilities") {
+        it("should trim spaces from label value") {
+          val plugin = new CapabilitiesPlugin
+          val appSpec = mock[ApplicationSpec]
+          val taskInfoBuilder = TaskInfo.newBuilder
+
+          taskInfoBuilder.setCommand(CommandInfo.newBuilder()
+            .setValue("./a_command"))
+          taskInfoBuilder.getLabelsBuilder.addLabels(Label.newBuilder()
+            .setKey("MESOS_BOUNDING_CAPABILITIES")
+            .setValue("   NET_RAW  ;   NET_ADMIN  "))
+
+          plugin.taskInfo(appSpec, taskInfoBuilder)
+
+          assert(taskInfoBuilder.getContainer.getLinuxInfo.getBoundingCapabilities.getCapabilitiesList == List(
+            CapabilityInfo.Capability.NET_RAW,
+            CapabilityInfo.Capability.NET_ADMIN
+          ).asJava)
+        }
+
+        it("should not crash on unknown capabilities") {
+          val plugin = new CapabilitiesPlugin
+          val appSpec = mock[ApplicationSpec]
+          val taskInfoBuilder = TaskInfo.newBuilder
+
+          taskInfoBuilder.setCommand(CommandInfo.newBuilder()
+            .setValue("./a_command"))
+          taskInfoBuilder.getLabelsBuilder.addLabels(Label.newBuilder()
+            .setKey("MESOS_BOUNDING_CAPABILITIES")
+            .setValue("CUSTOM_CAPABILITY;UNKNOWN_CAPA"))
```

Comment:
I think it should crash the task (not start it) otherwise typo might not get caught

Reply:
Agreed.

Reply:
Fixed, I eventually decided to modify the task to write an error message in stderr and fail instead of throwing an exception in Marathon because it induced a weird behavior in the deployment and did not bring any value in the UI anyway.